From 23bdb52b3a5cf395dc3fc69cf1900685d140f682 Mon Sep 17 00:00:00 2001
From: Yair Shachar <Yair.Shachar@amd.com>
Date: Wed, 10 Feb 2016 20:47:59 +0200
Subject: [PATCH 1706/2094] VRAM Memory Eviction: first phase:

Getting the PDD from the BO.
Adding  an interface to get mapped size-.

Change-Id: I096c03c4454a83face9dd23595e3ddfa920dedaf
Signed-off-by: Yair Shachar <Yair.Shachar@amd.com>
Signed-off-by: Moses Reuben <moses.reuben@amd.com>

Conflicts:
	drivers/gpu/drm/amd/amdgpu/amdgpu.h
	drivers/gpu/drm/amd/amdgpu/amdgpu_amdkfd_gfx_v8.c
	drivers/gpu/drm/amd/amdgpu/amdgpu_amdkfd_gfx_v8_tonga.c
	drivers/gpu/drm/amd/amdkfd/kfd_chardev.c
	drivers/gpu/drm/amd/amdkfd/kfd_module.c
	drivers/gpu/drm/amd/amdkfd/kfd_process.c
	drivers/gpu/drm/amd/include/kgd_kfd_interface.h
	drivers/gpu/drm/radeon/radeon_kfd.c
---
 drivers/gpu/drm/amd/amdgpu/amdgpu.h | 2 ++
 drivers/gpu/drm/radeon/radeon.h     | 1 +
 2 files changed, 3 insertions(+)

diff --git a/drivers/gpu/drm/amd/amdgpu/amdgpu.h b/drivers/gpu/drm/amd/amdgpu/amdgpu.h
index c463152..274b2d8 100644
--- a/drivers/gpu/drm/amd/amdgpu/amdgpu.h
+++ b/drivers/gpu/drm/amd/amdgpu/amdgpu.h
@@ -166,6 +166,7 @@ struct amdgpu_cs_parser;
 struct amdgpu_job;
 struct amdgpu_irq_src;
 struct amdgpu_fpriv;
+struct kfd_process_device;
 
 enum amdgpu_cp_irq {
 	AMDGPU_CP_IRQ_GFX_EOP = 0,
@@ -433,6 +434,7 @@ struct amdgpu_bo {
 	struct amdgpu_mn		*mn;
 	struct list_head		mn_list;
 	struct list_head		shadow_list;
+	struct kfd_process_device	*pdd;
 };
 #define gem_to_amdgpu_bo(gobj) container_of((gobj), struct amdgpu_bo, gem_base)
 
diff --git a/drivers/gpu/drm/radeon/radeon.h b/drivers/gpu/drm/radeon/radeon.h
index 99bb37b..cf60801 100644
--- a/drivers/gpu/drm/radeon/radeon.h
+++ b/drivers/gpu/drm/radeon/radeon.h
@@ -512,6 +512,7 @@ struct radeon_bo {
 
 	struct radeon_mn		*mn;
 	struct list_head		mn_list;
+	struct kfd_process_device	*pdd;
 };
 #define gem_to_radeon_bo(gobj) container_of((gobj), struct radeon_bo, gem_base)
 
-- 
2.7.4

